farasa/__base.py

```diff
@@ -15,43 +15,50 @@
 
 class FarasaBase:
     task = None
-    __base_dir = Path(__file__).parent.absolute()
-    __bin_dir = Path(f"{__base_dir}/farasa_bin")
-    __bin_lib_dir = Path(f"{__bin_dir}/lib")
+    base_dir = Path(__file__).parent.absolute()
+    bin_dir = Path(f"{base_dir}/farasa_bin")
+    bin_lib_dir = Path(f"{bin_dir}/lib")
 
     # shlex not compatible with Windows replace it with list()
     # set java encoding with option `-Dfile.encoding=UTF-8`
-    __BASE_CMD = ["java", "-Dfile.encoding=UTF-8", "-jar"]
-    __APIs = {
-        "segment": __BASE_CMD + [str(__bin_lib_dir / "FarasaSegmenterJar.jar")],
-        "stem": __BASE_CMD
-        + [str(__bin_lib_dir / "FarasaSegmenterJar.jar"), "-l", "true"],
-        "NER": __BASE_CMD + [str(__bin_dir / "FarasaNERJar.jar")],
-        "POS": __BASE_CMD + [str(__bin_dir / "FarasaPOSJar.jar")],
-        "diacritize": __BASE_CMD + [str(__bin_dir / "FarasaDiacritizeJar.jar")],
-    }
-    __interactive = False
-    __task_proc = None
+    BASE_CMD = ["java", "-Dfile.encoding=UTF-8", "-jar"]
+    # __APIs = {
+    # "segment": __BASE_CMD + [str(__bin_lib_dir / "FarasaSegmenterJar.jar")],
+    # "stem": __BASE_CMD
+    # + [str(__bin_lib_dir / "FarasaSegmenterJar.jar"), "-l", "true"],
+    # "NER": __BASE_CMD + [str(__bin_dir / "FarasaNERJar.jar")],
+    # "POS": __BASE_CMD + [str(__bin_dir / "FarasaPOSJar.jar")],
+    # "diacritize": __BASE_CMD + [str(__bin_dir / "FarasaDiacritizeJar.jar")],
+    # }
+    interactive = False
+    task_proc = None
     logger = None
 
+    @property
+    def command(self):
```

Comment:
I like this approach to adding paths to personal JAR files. I recommend three things: 
1. Calling this function in the init function to check if user has overridden this method or not. If user has not overridden, it would raise the NotImplemented error.
2. Add more documentation to the error message for beginner users to understand how to initialize their own services from Farasa JAR files.
3. Add Spell Checking service as an example to follow, especially locating and routing to files, or including the JARs within the bin folder of Farasa.

Reply:
Thanks for this review, @abdullah-shwaiky, 

- for the first point, actually, the command property was not intended for the end users. Instead, it is the responsibility of the maintainers who are adding a new task to verify that this property has been overridden which will be failing once they do testing anyways. However, it is important to let the users know if they need to give the JAR file for a task that we cannot support out of the box in farasapy. This is why I added this check here [https://github.com/MagedSaeed/farasapy/blob/9305d5bf74c44cc100e41d94b5d2bf0c6ebbc507/farasa/__base.py#L37]. By the way, it seems that you did not review the latest commit of the branch and hence this comment.

- for the second and third point, I think it is important to add documentation in the README about the supported tasks out of the box and the tasks that needs JAR files. This will happen, InshaAllah, after we merge the SpellCheck PR as an example of this kind of tasks.

Reply:
Hi @MagedSaeed 
I apologize for the slip up, I must have assumed commits are ordered counter-chronologically. I'll be more cautious next time.

Thank you for clearing that up. The new commit looks even cleaner and the error message is well documented. Great work!